source/service_test.go

```diff
@@ -2130,7 +2272,7 @@ func TestHeadlessServicesHostIP(t *testing.T) {
 			false,
 		},
 		{
-			"annotated Headless services return endpoints for pods missing hostname",
+			"annotated Headless services with suppressHeadlessRoot return endpoints for pods missing hostname",
```

Comment:
with suppressHeadlessRoot means you would need to set it to true but it's false.

Reply:
I didn't get the comment at first.  You're right.  I've split this into two test cases, and updated the naming of all of the tests to be more clear.